### ISSUE TITLE
crux.eval of a parametric fem.solve ignores in-place parameter module mutation

### ISSUE THREAD

**Issue by armbrusl** [FhG-IISB/jNO#100]
## What

Evaluating a parametric `fem.solve()` node via `crux.eval` after mutating the parameter's stored module in place (`eqx.tree_at`, the documented data-field idiom from `tests/test_domain_decomposition.py`) returns the value at the **old** parameter — silently.

## Reproduction (isolated 2026-08-15)

```python
g = jno.np.parameter((1,), name="g", key=jax.random.PRNGKey(0))
fem = jno.fem([ui.x*vi.x + ui.y*vi.y - 1.0*vi, u(xb, yb) - g])   # parametric (Dirichlet value)
crux = jno.core([(fem.solve()).mse], domain=d)

def at(val):
    g.model.module = eqx.tree_at(lambda m: m.value, g.model.module, jnp.asarray([val]))
    return float(np.asarray(crux.eval([fem.solve()])).reshape(-1).mean())

at(1.0)   # 0.0220
at(5.0)   # 0.0220  <-- identical to the last bit; the mutation was ignored
```

Training through `crux.solve` works (the same session's boundary-value recovery tests hit <1% on three solve paths), so the staleness is specific to the **eval-after-mutation** path — either an eval-side cache keyed on the expression, or a captured module snapshot.

## Why it matters

This is the silent-wrong class: a user sweeping a parameter by mutation + eval gets a flat response and no error. It also cost a debugging round in this session (a gradient test read slope = 0.000 and looked like a broken feature).

Either honor the live module at eval time, or refuse/warn when a registered parameter's module has changed since compile. A test pinning whichever contract is chosen belongs next to the DD data-field tests.

**Comment by armbrusl** [comment]
Fixed in #117. The cause is not an eval-side cache, and it is not specific to `fem.solve()`.

`jno.core(...)` reads `layer.module` once at construction, through `build_single_layer_params`, and evaluates its own copy from then on. `crux.models` holds no reference to `g.model`, so a later in-place swap is invisible. Evaluating the parameter directly shows it plainly — the core never saw 1 or 5, it saw the value `g` held when the core was built:

```
g=1: stored=[1.]  eval(g*1.0)=[0.]  sol.mean=0.022491
g=5: stored=[5.]  eval(g*1.0)=[0.]  sol.mean=0.022491
```

That also explains why the DD data-field tests work: they mutate and then call `fem.solve()` / `fdm.solve()` **eagerly**, with no core in the path, so the module is read at solve time.

Of the two options in the issue, refuse is the right one. Honouring the live module in `eval` would patch only `eval` — `self.models`, what training starts from and what every other read uses, would still hold the old value, so eval and solve would then disagree about the same parameter; and it would skip the build a module goes through on the way in (pretrained weights, a callable initializer, an initialize mask, a dtype cast, sharding).

One subtlety worth recording: `solve()` syncs trained weights back onto `layer.module`, so a check against the module the core was *built* from refuses every eval that follows a solve. The check tracks the module each layer carried when the core **last** read it, refreshed at that write-back.

**Adjacent, not fixed in #117:** `crux.solve()` starts from `dict(self.models)`, the same snapshot, so a module swapped before solve is silently ignored there too. `tests/test_integration.py:289` installs a paramax wrapper after its core is built and then asserts training yields a finite loss — the wrapper never reaches the forward path, so that test passes whether or not auto-unwrapping works. Extending the guard to `solve` would surface it, but it needs that test restructured, so it belongs in its own change.
